packages/template-retail-react-app/app/components/item-variant/item-price.jsx

```diff
@@ -43,74 +36,68 @@ PricePerItem.propTypes = {
  */
 const ItemPrice = ({currency, align = 'right', baseDirection = 'column', ...props}) => {
     const variant = useItemVariant()
-    const {data: basket} = useCurrentBasket()
-    const {currency: activeCurrency} = useCurrency()
-    const intl = useIntl()
-
-    const {price, basePrice, priceAfterItemDiscount} = variant
-    const isProductASet = variant?.type?.set
-
-    const displayPrice =
-        typeof priceAfterItemDiscount === 'number' ? Math.min(price, priceAfterItemDiscount) : price
-
-    const hasDiscount = displayPrice !== price
-
+    const {price, priceAfterItemDiscount} = variant
+    const isASet = variant?.type?.set
+    const isMaster = variant?.type?.master
+    let priceData
+    // this indicates the variant is on cart page
+    // when on cart page, the variant data is merged with basket item info,
+    // it will have price info that takes quantity and different promotions into the account
+    // we should prioritize use these values for display price
+    // and should NOT attempt to re-calculate at all cost here
```

Comment:
Can you take another stab at making this more condensed and concise. Remember that because of extensibility every component is part of our public API, and they will be reading these comments. 

Aside: I think you meant to say "into account" and not "into the account"